src/components/Dropdown/Dropdown.tsx

```diff
@@ -66,8 +67,17 @@ export function Dropdown(props: PropsWithChildren<DropdownProps>): JSX.Element {
     alwaysSelectOption = false
   } = props;
 
+  //reset the id when the component is re-rendered
```

Comment:
we should pull this logic out into a custom hook. in that hook, we can first check if we have `useId` available through react 18 and use that before falling back to this logic if it's not present